### ISSUE TITLE
Use `shares` instead of `assets` to repay max

### ISSUE THREAD

**Issue by MerlinEgalite** [OasisDEX/oasis-earn-sc#591]
https://github.com/OasisDEX/oasis-earn-sc/blob/5a2c7fef7186e1f0bc186d2f717c8b7a006ff532/packages/dma-contracts/contracts/actions/morpho-blue/Payback.sol#L35

Using assets to repay the full debt can leave some shares on the protocol and thus no entirely close the position.

See this snippet as example:
https://github.com/morpho-org/morpho-blue-snippets/blob/190ced2d6cee553e4fd117e4fc3adf7ab3e6b4fb/src/morpho-blue/MorphoBlueSnippets.sol#L377-L383

**Comment by Jean-Grimal** [comment]
Yes I would recommend to use the shares input to repay all when `paybackData.paybackAll == true`.
